src/mds/Server.cc

```diff
@@ -1285,9 +1285,8 @@ void Server::journal_close_session(Session *session, int state, Context *on_safe
   mdlog->flush();
 
   // clean up requests, too
-  for (auto p = session->requests.begin(); !p.end(); ) {
-    MDRequestRef mdr(*p);
-    ++p;
+  while(!(session->requests.empty())) {
```

Comment:
nit: inner parens are redundant. Better would just be:

`while (!session->requests.empty()) {`

Reply:
done.Thanks